File: CS/WindowsApplication3/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using System.Threading;
using DevExpress.XtraEditors;



namespace DXSample {
    public partial class Form1: Form {
        public Form1() {
            InitializeComponent();
           
        }
       
        private void Form1_Load(object sender, EventArgs e) {
            radioGroup1.SelectedIndex = 2;
        }

        private static Product CreateProduct() {
            Product pr = new Product("123-12-23", "Orange", "Fruit", 12, 100);
            return pr;
        }

    
        private void radioGroup1_SelectedIndexChanged(object sender, EventArgs e) {
            RadioGroup radioGroup = sender as RadioGroup;
            string cultureName = radioGroup.EditValue.ToString();

            SetCulture(cultureName);
           
            propertyGridControl1.SelectedObject = null;
            propertyGridControl1.Rows.Clear();
            propertyGridControl1.SelectedObject = CreateProduct();
        }

        private static void SetCulture(string cultureName) {
            Thread.CurrentThread.CurrentCulture = new System.Globalization.CultureInfo(cultureName);
            Thread.CurrentThread.CurrentUICulture = new System.Globalization.CultureInfo(cultureName);
        }
    }
    
    public class Product {
        string productCode, name, category; 
        decimal price;
        int quantity;

        public Product(string code, string name, string category, decimal price, int quantity) {
            this.productCode = code;
            this.name = name;
            this.category = category;
            this.price = price;
            this.quantity = quantity;
        }
        [CustomDisplayNameAttribute("ProductCode")]
        public string ProductCode {
            get { return productCode; }
            set { productCode = value; }
        }
        [CustomDisplayNameAttribute("Name")]
        public string Name {
            get { return name; }
            set { name = value; }
        }
        [CustomDisplayNameAttribute("Category")]
        public string Category {
            get { return category; }
            set { category = value; }
        }
        [CustomDisplayNameAttribute("Price")]
        public decimal Price {
            get { return price; }
            set { price = value; }
        }
        [CustomDisplayNameAttribute("Quantity")]
        public int Quantity {
            get { return quantity; }
            set { quantity = value; }
        }
    }

    public class CustomDisplayNameAttribute: DisplayNameAttribute {
        public CustomDisplayNameAttribute(string displayName): base(displayName) { }
        public override string DisplayName {
            get {
                return CustomLocalizer.GetLocalizedString(base.DisplayName);
            }
        }
    }


    public static class CustomLocalizer{
        public static string GetLocalizedString(string name){
            if(Thread.CurrentThread.CurrentCulture.Name == "fr-FR") {
                if(name == "ProductCode")
                    return "Produit Code";
                if(name == "Name")
                    return "Nom";
                if(name == "Category")
                    return "Catégorie";
                if(name == "Price")
                    return "Prix";
                if(name == "Quantity")
                    return "Quantité";
            }
            else
                if(Thread.CurrentThread.CurrentCulture.Name == "de-DE") {
                    if(name == "ProductCode")
                        return "Produkt Code";
                    if(name == "Name")
                        return "Name";
                    if(name == "Category")
                        return "Kategorie";
                    if(name == "Price")
                        return "Preis";
                    if(name == "Quantity")
                        return "Zahl";
                }
            return name;
        }

    }
}
       
  

  

File: VB/WindowsApplication3/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports System.Threading
Imports DevExpress.XtraEditors



Namespace DXSample
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()

		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			radioGroup1.SelectedIndex = 2
		End Sub

		Private Shared Function CreateProduct() As Product
			Dim pr As New Product("123-12-23", "Orange", "Fruit", 12, 100)
			Return pr
		End Function


		Private Sub radioGroup1_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles radioGroup1.SelectedIndexChanged
			Dim radioGroup As RadioGroup = TryCast(sender, RadioGroup)
			Dim cultureName As String = radioGroup.EditValue.ToString()

			SetCulture(cultureName)

			propertyGridControl1.SelectedObject = Nothing
			propertyGridControl1.Rows.Clear()
			propertyGridControl1.SelectedObject = CreateProduct()
		End Sub

		Private Shared Sub SetCulture(ByVal cultureName As String)
			Thread.CurrentThread.CurrentCulture = New System.Globalization.CultureInfo(cultureName)
			Thread.CurrentThread.CurrentUICulture = New System.Globalization.CultureInfo(cultureName)
		End Sub
	End Class

	Public Class Product
		Private productCode_Renamed, name_Renamed, category_Renamed As String
		Private price_Renamed As Decimal
		Private quantity_Renamed As Integer

		Public Sub New(ByVal code As String, ByVal name As String, ByVal category As String, ByVal price As Decimal, ByVal quantity As Integer)
			Me.productCode_Renamed = code
			Me.name_Renamed = name
			Me.category_Renamed = category
			Me.price_Renamed = price
			Me.quantity_Renamed = quantity
		End Sub
		<CustomDisplayNameAttribute("ProductCode")> _
		Public Property ProductCode() As String
			Get
				Return productCode_Renamed
			End Get
			Set(ByVal value As String)
				productCode_Renamed = value
			End Set
		End Property
		<CustomDisplayNameAttribute("Name")> _
		Public Property Name() As String
			Get
				Return name_Renamed
			End Get
			Set(ByVal value As String)
				name_Renamed = value
			End Set
		End Property
		<CustomDisplayNameAttribute("Category")> _
		Public Property Category() As String
			Get
				Return category_Renamed
			End Get
			Set(ByVal value As String)
				category_Renamed = value
			End Set
		End Property
		<CustomDisplayNameAttribute("Price")> _
		Public Property Price() As Decimal
			Get
				Return price_Renamed
			End Get
			Set(ByVal value As Decimal)
				price_Renamed = value
			End Set
		End Property
		<CustomDisplayNameAttribute("Quantity")> _
		Public Property Quantity() As Integer
			Get
				Return quantity_Renamed
			End Get
			Set(ByVal value As Integer)
				quantity_Renamed = value
			End Set
		End Property
	End Class

	Public Class CustomDisplayNameAttribute
		Inherits DisplayNameAttribute
		Public Sub New(ByVal displayName As String)
			MyBase.New(displayName)
		End Sub
		Public Overrides ReadOnly Property DisplayName() As String
			Get
				Return CustomLocalizer.GetLocalizedString(MyBase.DisplayName)
			End Get
		End Property
	End Class


	Public NotInheritable Class CustomLocalizer
		Private Sub New()
		End Sub
		Public Shared Function GetLocalizedString(ByVal name As String) As String
			If Thread.CurrentThread.CurrentCulture.Name = "fr-FR" Then
				If name = "ProductCode" Then
					Return "Produit Code"
				End If
				If name = "Name" Then
					Return "Nom"
				End If
				If name = "Category" Then
					Return "Catégorie"
				End If
				If name = "Price" Then
					Return "Prix"
				End If
				If name = "Quantity" Then
					Return "Quantité"
				End If
			Else
				If Thread.CurrentThread.CurrentCulture.Name = "de-DE" Then
					If name = "ProductCode" Then
						Return "Produkt Code"
					End If
					If name = "Name" Then
						Return "Name"
					End If
					If name = "Category" Then
						Return "Kategorie"
					End If
					If name = "Price" Then
						Return "Preis"
					End If
					If name = "Quantity" Then
						Return "Zahl"
					End If
				End If
			End If
			Return name
		End Function

	End Class
End Namespace





